src/rest-server/test/v2/protocol.js

```diff
@@ -0,0 +1,470 @@
+    completion:
+      minSucceededTaskCount: 1
+    dockerImage: caffe_example
+    resourcePerInstance:
+      cpu: 4
+      memoryMB: 8192
+      gpu: 1
+    commands:
+      - ./examples/mnist/train_lenet.sh
+  `,
+
+  'should have required property \'taskRoles\'': `
+protocolVersion: 2
+name: caffe_mnist
+type: job
+version: 1.0.0
+contributor: OpenPAI
+prerequisites:
+  - protocolVersion: 2
+    name: caffe_example
+    type: dockerimage
+    version: 1.0.0
+    contributor : OpenPAI
+    description: caffe
+    uri: openpai/pai.example.caffe
+  `,
+
+  'should have required property \'name\'': `
+protocolVersion: 2
+type: job
+version: 1.0.0
+contributor: OpenPAI
+prerequisites:
+  - protocolVersion: 2
+    name: caffe_example
+    type: dockerimage
+    version: 1.0.0
+    contributor : OpenPAI
+    description: caffe
+    uri: openpai/pai.example.caffe
+taskRoles:
+  valid_name:
+    instances: 1
+    completion:
+      minSucceededTaskCount: 1
+    dockerImage: caffe_example
+    resourcePerInstance:
+      cpu: 4
+      memoryMB: 8192
+      gpu: 1
+    commands:
+      - ./examples/mnist/train_lenet.sh
+  `,
+
+  'should have required property \'gpu\'': `
+protocolVersion: 2
+name: caffe_example
+type: job
+version: 1.0.0
+contributor: OpenPAI
+prerequisites:
+  - protocolVersion: 2
+    name: caffe_example
+    type: dockerimage
+    version: 1.0.0
+    contributor : OpenPAI
+    description: caffe
+    uri: openpai/pai.example.caffe
+taskRoles:
+  valid_name:
+    instances: 1
+    completion:
+      minSucceededTaskCount: 1
+    dockerImage: caffe_example
+    resourcePerInstance:
+      cpu: 4
+      memoryMB: 8192
+    commands:
+      - ./examples/mnist/train_lenet.sh
+  `,
+
+  'should have required property \'commands\'': `
+protocolVersion: 2
+name: caffe_example
+type: job
+version: 1.0.0
+contributor: OpenPAI
+prerequisites:
+  - protocolVersion: 2
+    name: caffe_example
+    type: dockerimage
+    version: 1.0.0
+    contributor : OpenPAI
+    description: caffe
+    uri: openpai/pai.example.caffe
+taskRoles:
+  valid_name:
+    instances: 1
+    completion:
+      minSucceededTaskCount: 1
+    dockerImage: caffe_example
+    resourcePerInstance:
+      cpu: 4
+      memoryMB: 8192
+      gpu: 1
+  `,
+
+  'should be equal to one of the allowed values': `
+protocolVersion: 2
+name: caffe_mnist
+type: job
+version: 1.0.0
+contributor: OpenPAI
+prerequisites:
+  - protocolVersion: 2
+    name: caffe_example
+    type: dockerimage
+    version: 1.0.0
+    contributor : OpenPAI
+    description: caffe
+  `,
+};
+
+// unit tests for protocol in API v2
+describe('API v2 Unit Tests: protocol', () => {
+  // protocol schema test
+  it('test protocol schema for valid protocol', (done) => {
+    for (let validProtocolYAML of Object.values(validProtocolYAMLs)) {
+      const validProtocolJSON = yaml.safeLoad(validProtocolYAML);
+      expect(protocolSchema.validate(validProtocolJSON)).to.be.true;
+    }
+    done();
```

Comment:
No need to `done` (both param and call here) since it's a sync test case.

Reply:
updated